core/src/main/java/com/datastax/oss/driver/internal/core/config/typesafe/DefaultProgrammaticDriverConfigLoaderBuilder.java

```diff
@@ -37,7 +37,9 @@
     implements ProgrammaticDriverConfigLoaderBuilder {
 
   public static final Supplier<Config> DEFAULT_FALLBACK_SUPPLIER =
-      () -> ConfigFactory.defaultApplication().withFallback(ConfigFactory.defaultReference());
+      () ->
+          ConfigFactory.defaultApplication()
+              .withFallback(ConfigFactory.defaultReference(DriverConfigLoader.DRIVER_CLASS_LOADER));
```

Comment:
This should fix the programmatic config case where users will provide overrides programmatically for all settings that should override the Driver's defaults, and should correctly be able to access the Driver's defaults by specifying its class loader.

Reply:
Hmm I think this isn't enough. You need to use `defaultApplication(java.lang.ClassLoader)` instead: 

```
ConfigFactory.defaultApplication(DriverConfigLoader.DRIVER_CLASS_LOADER)
              .withFallback(ConfigFactory.defaultReference(DriverConfigLoader.DRIVER_CLASS_LOADER))
```

Reply:
I did think about doing that, but in a web environment, it may be more limiting. If we do not provide a ClassLoader to TypeSafe's `ConfigFactory.defaultApplication()`, it will use the current Thread's Context ClassLoader. Please correct me where I'm wrong:
1) In a normal Java application, the class loaders are all the same as the context class loader.
2) In a web environment, the context class loader should be able to load resources from the app bundle ("WEB-INF/lib") as well as the web container ("/lib"). If we specify the Driver's ClassLoader here, then we could be limiting the ability to find application config if the Driver jar is in the web container's system folder ("/lib"), and the app jar is in the app folder ("WEB-INF/lib").
3) In an OSGi container, neither of these will work for finding application config in the application bundle as the Context ClassLoader isn't the ClassLoader for the application, and the Driver's ClassLoader is in a different bundle.

Also, I don't think we really need to look for Application config specifically in the Driver's ClassLoader since the Driver doesn't ship with an "application.conf". It only works if the Driver's ClassLoader is the ClassLoader for the app jar (or is a child of the app jar's ClassLoader).

Reply:
You are absolutely right 👍